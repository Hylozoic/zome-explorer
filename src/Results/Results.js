import React, { useState } from 'react'
import CallRecord from '../CallRecord'

export default function Results ({ history }) {
  const [filter, setFilter] = useState('')
  const filteredRecords = history.filter(callRecord => JSON.stringify(callRecord).indexOf(filter) >= 0)

return <div className='results'>
    <div>
      <label className='input-label'>Filter</label>
      <input value={filter} onChange={({ target: { value }}) => setFilter(value)} />
    </div>
    {filteredRecords.map((callRecord, i) => <CallRecord callRecord={callRecord} key={i} />)}
  </div>
}