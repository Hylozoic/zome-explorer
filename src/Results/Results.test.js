import Results from './Results'
import { shallow } from 'enzyme'
import React from 'react'

it('matches the last snapshot', () => {
  const props = {
    history: [
      {name: "one call record"},
      {name: "another call record"}
    ]
  }  
  const wrapper = shallow(<Results {...props} />)
  expect(wrapper).toMatchSnapshot()
})
