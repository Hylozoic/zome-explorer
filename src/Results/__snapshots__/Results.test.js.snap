// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the last snapshot 1`] = `
<div
  className="results"
>
  <div
    className="filter-row"
  >
    <label
      className="input-label"
    >
      Filter
    </label>
    <input
      onChange={[Function]}
      value=""
    />
  </div>
  <CallRecord
    callRecord={
      Object {
        "name": "one call record",
      }
    }
    key="0"
  />
  <CallRecord
    callRecord={
      Object {
        "name": "another call record",
      }
    }
    key="1"
  />
</div>
`;
